src/Array/Extra.elm

```diff
@@ -236,20 +236,17 @@ and collect the result in a `List`.
             >> Html.div []
 
 -}
-indexedMapToList :
-    (Int -> element -> mappedElement)
-    -> (Array element -> List mappedElement)
-indexedMapToList mapIndexedElement =
-    \array ->
-        array
-            |> Array.foldr
-                (\element ( i, listSoFar ) ->
-                    ( i - 1
-                    , listSoFar |> (::) (mapIndexedElement i element)
-                    )
+indexedMapToList : (Int -> a -> b) -> (Array a -> List b)
+indexedMapToList mapIndexedElement array =
```

Comment:
`mapIndexedElement` to just `fn` or `mapper`? "map" is ambiguously both a noun and a verb, although the signature should make it clear.